sympy/functions/special/bsplines.py

```diff
@@ -331,7 +331,7 @@ def interpolating_spline(d, x, X, Y):
     A = [[b.subs(x, v) for b in basis] for v in X]
 
     coeff = linsolve((Matrix(A), Matrix(Y)), symbols("c0:{}".format(len(X)), cls=Dummy))
-    coeff = list(coeff)[0]
+    coeff = next(iter(coeff))
```

Comment:
I would just write
```
[coeff] = linsolve(...)
```
I prefer this form because it asserts that there is only one element.